docs/how-to/push-update-on-mainnet.md

```diff
@@ -1,14 +1,62 @@
 Overview
 ========
 
-Software update is a protocol mechanism that allows nodes to agree on software changes without altering protocol constants and update to this software version.
+Software update is a protocol mechanism that allows nodes to agree on
+software changes without altering protocol constants and update to
+this software version.
+
+Protocol update is a different protocol mechanism that allows nodes to
+agree on protocol changes without breaking compatibility with older
+versions. Protocol update must be done with software update
+(i. e. it's impossible to make a protocol update without a software
+update). Note, however, that software update can be done for only one
+software. Usually it will be `cardano-sl` which is used by core and
+relay nodes and differs from `csl-daedalus` which is used by end users.
+
+Currently we support two different applications (from the standpoint
+of update system): `cardano-sl` and `csl-daedalus`. The former is used
+by core and relay nodes, the latter is used by end-users
+(wallets). One of the goals of update system is to provide a way to
+know whether a particular software is approved by major stakeholders
+by looking at blockchain. For `csl-daedalus` it makes perfect sense:
+from blockchain we know hashes of installers which are approved. But
+for `cardano-sl` it doesn't make sense, because we don't even have a
+notion of _installer_ for it. So when we update `cardano-sl` it
+doesn't make sense to attach any data about installers into update
+proposal. There are few reasons to propose an update of `cardano-sl`:
+1. It allows us to make a protocol update without updating end
+   users. For example we may want to change some protocol parameter
+   and not bother users with updates. Note that we can also propose an
+   update of non-existing application, but it doesn't make much sense
+   to pollute the namespace.
+2. We can permanently register updates of software used by core nodes
+   in blockchain to be able to review this history in future (not
+   essential, probably doesn't deserve spending time on it).
+
+Flow for an update is the following (brief sketch, no details):
 1. Software update is proposed. Proposal is a datatype that gets into
    the blockchain. It contains information about version changes and
-   hashes of update files.
-2. Update files are uploaded to the S3 bucket.
-3. Software update is confirmed by voting from majority of nodes.
-4. Nodes that see an update try to download and apply it
-   automatically.
+   hashes of update files (can be empty). It also contains
+   `BlockVersion` and `BlockVersionData`. They define protocol version
+   and its parameters. They can be the same as the previous ones (in
+   which case there is no protocol update) or can differ (then there
+   is also a protocol update).
+2. New software is prepared, some values in configuration are
```

Comment:
This should be step #1 -- the protocol proposal can't happen without the installers, unless you're OK with performing two separate proposals (for protocol, then for installers).